### ISSUE TITLE
Scraping order different each time

### ISSUE THREAD

**Issue by hallerio** [rchipka/node-osmosis#116]
Hey,

I'm trying to scrape a news section, but the quirk is that the news don't have a date displayed, so I really need to have the same order in the result array as on the page. For each article I'm following the link to get the full text.

I've set the concurrency to 1 in the config but the scrape order seems completely random and different each time. Is there a way to prevent that? Here's my code, maybe I'm doing something wrong:

```
var news = [];

osmosis
.get('__URL__?page=0')
.config('concurrency', 1)
.paginate('.pager-next a')
.find('#content-area .news-teaser')
.set({
    'title': 'a.kiemelt02',
    'image': 'img@src',
    'text': [
        osmosis
        .follow('a.kiemelt02@href')
        .find('#content-area .news-teaser-txt p')
    ]
})
.data(function(element) {
    news.push(element);
})
.done(function() {
    jsonfile.writeFileSync(file, news);
});
```

**Comment by balinthaller** [comment]
After further investigation, only the follow method call messes up the order. Without following the links, the order remains intact.

**Comment by rchipka** [comment]
Try making the .set object an array and see if the problem persists.

```javascript
osmosis
...
.set([
        {
            'title': 'a.kiemelt02',
            'image': 'img@src',
            'text': [
                osmosis
                .follow('a.kiemelt02@href')
                .find('#content-area .news-teaser-txt p')
            ]
        }
    ])
.data(function(array) {
    jsonfile.writeFileSync(file, array);
})
```

**Comment by balinthaller** [comment]
Unfortunately that doesn't help, all it does is put each entry in a separate, one item array.

**Comment by zxzlsh** [comment]
Try that.

```

var news = [];
var count = 0;

osmosis
    .get('__URL__?page=0')
    .config('concurrency', 1)
    .paginate('.pager-next a')
    .find('#content-area .news-teaser')
    .set({
        'title': 'a.kiemelt02',
        'image': 'img@src',
    })
    .data(function (element) {
        news.push(element);
    })
    .done(function () {

        news.forEach(function (item) {
            FollowTheLink(item)
        }, this);

    });

function FollowTheLink(web) {
    osmosis.get(web.link)
        .find('#content-area .news-teaser-txt')
        .set({
            'text': 'p',
        })
        .data(function (item) {
            web.text = item.text;
        })
        .done(function () {

            count++;

            if (count === news.length) {

                jsonfile.writeFileSync(file, news);

            }
        })
}

```

**Comment by rchipka** [comment]
This should be fixed in 0.1.5
